### PR TITLE
Make it an error (not a deprecation warning) to reference tasks from an included build using finalizedBy, mustRunAfter, or shouldRunAfter

### LINKED ISSUE #15875: Turn 'Using mustRunAfter/shouldRunAfter/finalizedBy to reference tasks from another build' into an error
### Summary
Turn nagging in `TaskNode.deprecateLifecycleHookReferencingNonLocalTask()` into an error.

### Checklist
- [ ] Update the upgrade guide
- [ ] Check User Manual for dangling references and outdated documentation

### PR DESCRIPTION
**Pull request by tresat**
Fixes #15875

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]
LGTM

**Review by big-guy — APPROVED** [review]
Please add a note about this to the upgrading guide after https://github.com/gradle/gradle/pull/21478 is merged and master has been updated to 8.0

**Comment by tresat** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -33,6 +33,22 @@ Some plugins will break with this new version of Gradle, for example because the
 . Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
 . Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.
 
+[[changes_8.0]]
+== Upgrading from 7.6 and earlier
+
+=== Warnings that are now errors
+
+==== Referencing tasks in an included build with `finalizedBy`, `mustRunAfter` or `shouldRunAfter`
+If you use `finalizedBy`, `mustRunAfter` or `shouldRunAfter` to reference tasks contained in an included build, this will now result in an execution time error upon running the referencing tasks.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:42]:
```suggestion
Referencing tasks contained in an included build with any of the following methods now results in an execution time error:

- `finalizedBy`
- `mustRunAfter`
- `shouldRunAfter`
```

Suggestion: move the list of methods into a bulleted list so it's easier to read the sentence on its own (and then look at the list of methods to see if you're affected).

Feel free to reword the introduction sentence if my reword isn't totally accurate.

**Review by nathan-contino — APPROVED** [review]
LGTM, one suggestion in the docs.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 1 above)
Comment (tresat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:42]: Thanks for the suggestion, I added it [here](https://github.com/gradle/gradle/pull/21511)

Comment (tresat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:42]: @lkasso - some docs feedback here

**Pull request merged**